Drop non-list HTML dependencies from exercise html_output

gadenbuie opened the pull request:
Fixes #484

PR task list:
- [x] Update NEWS
- [x] Add tests (if possible)
- [x] Update documentation with `devtools::document()`

schloerke commented on tests/testthat/test-exercise.R:
```diff
@@ -218,3 +218,27 @@ test_that("serialized exercises produce equivalent evaluate_exercise() results",
+    user_code = deparse(quote(
+      htmltools::tagList(
+        htmltools::tags$head(htmltools::tags$style(".leaflet-container {backround:#FFF}")),
+        learnr::html_dependency_jquery()
```
(Prolly from local copy/paste debugging)
```suggestion
        html_dependency_jquery()
```

schloerke commented on tests/testthat/test-exercise.R:
```diff
@@ -218,3 +218,27 @@ test_that("serialized exercises produce equivalent evaluate_exercise() results",
+  # But we keep the dependency that came from a pkg
+  expect_equal(
+    ex_res_html_deps[[1]],
+    learnr::html_dependency_jquery()
```
(Prolly from local copy/paste debugging)
```suggestion
    html_dependency_jquery()
```

gadenbuie commented:
I added another commit that doesn't seem to be showing up here https://github.com/gadenbuie/learnr/commit/b01bd2242516d6d954e1f35668ed069d51f81781

I also needed a `with_tempdir()` around the `render_exercise()`, and I changed the test to use a dependency defined inside learnr directly.

gadenbuie commented:
I'm not sure what went wrong with this PR, I'm going to close and reopen with the fixed up history
